api/envoy/config/route/v3/route_components.proto

```diff
@@ -833,6 +833,18 @@ message RouteAction {
       type.matcher.v3.RegexMatchAndSubstitute regex_rewrite = 2;
     }
 
+    // CookieAttribute defines an API for adding additional attributes for a HTTP cookie.
+    message CookieAttribute {
+      // The name of the cookie attribute.
+      string name = 1
+          [(validate.rules).string =
+               {min_len: 1 max_bytes: 16384 well_known_regex: HTTP_HEADER_NAME strict: false}];
```

Comment:
Not sure if the HTTP header name format is appropriate for the cookie attributes. If there are only some enumatable values, maybe `in` could be used here rather than `well_known_regex`.

Reply:
@wbpcode @mattklein123 any suggestions to take this forward?

Reply:
Can you figure out where in the RFC it specifies valid characters in cookie names? cc @RyanTheOptimist

Reply:
Cookie names follow `token` I guess same as Http Header names as per https://www.rfc-editor.org/rfc/rfc6265#section-4.1 and https://www.rfc-editor.org/rfc/rfc2616#section-2.2

Reply:
OK sounds good then seems like this is fine for now. Thanks for looking.